src/components/inputs/text/NameTokenSelector/NameTokenSelector.tsx

```diff
@@ -0,0 +1,590 @@
+  const [filteredTokens, setFilteredTokens] =
+    useState<
+      Array<{ id: string; name?: string; ticker?: string } | undefined>
+    >();
+  const [selectedToken, setSelectedToken] = useState<
+    { id: string; name: string; ticker: string } | undefined
+  >(undefined);
+  const [searching, setSearching] = useState<boolean>(false);
+  const [searchActive, setSearchActive] = useState(false);
+  const [validImport, setValidImport] = useState<boolean | undefined>(
+    undefined,
+  );
+
+  const listRef = useRef<HTMLDivElement>(null);
+  const [listPage, setListPage] = useState<number>(1);
+  const listItemCount = 5;
+
+  useEffect(() => {
+    selectedTokenCallback(
+      selectedToken ? new ArweaveTransactionID(selectedToken.id) : undefined,
+    );
+  }, [selectedToken]);
+
+  useEffect(() => {
+    getTokenList(walletAddress);
+  }, [walletAddress]);
+
+  useEffect(() => {
+    // token import update
+    if (searchText) {
+      handleTokenSearch(searchText);
+      setListPage(1);
+    }
+  }, [tokens]);
+
+  useEffect(() => {
+    if (!listRef.current) {
+      return;
+    }
+    document.addEventListener('mousedown', handleClickOutside);
+
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, [listRef]);
+
+  function handleClickOutside(e: any) {
+    e.preventDefault();
+    if (
+      listRef.current &&
+      e.target !== listRef.current &&
+      !listRef.current.contains(e.target)
+    ) {
+      setSearchText('');
+      setSearchActive(false);
+      setFilteredTokens(undefined);
+      setListPage(1);
+    }
+  }
+
+  async function getTokenList(
+    address: ArweaveTransactionID,
+    imports?: ArweaveTransactionID[],
+  ) {
+    try {
+      setLoading(true);
+      const fetchedIds = await arweaveDataProvider.getContractsForWallet(
+        pdnsSourceContract.approvedANTSourceCodeTxs.map(
+          (sourceCode) => new ArweaveTransactionID(sourceCode),
+        ),
+        address,
+      );
+      if (!fetchedIds.ids) {
+        throw new Error(
+          'Unable to find any Name Tokens for the provided address',
+        );
+      }
+      if (imports) {
+        imports.map((id) =>
+          arweaveDataProvider
+            .validateTransactionTags({
+              id: id.toString(),
+              requiredTags: {
+                'Contract-Src': pdnsSourceContract.approvedANTSourceCodeTxs,
+              },
+            })
+            .catch(() => {
+              if (searchText) {
+                setValidImport(false);
+              }
+              throw new Error('Invalid ANT Contract.');
+            }),
+        );
+        setValidImport(true);
+      }
+      const ids = fetchedIds.ids.concat(imports ?? []);
+
+      const contracts: [ArweaveTransactionID, PDNTContractJSON][] =
+        await Promise.all(
+          ids.map(async (id) => {
+            const state = await arweaveDataProvider.getContractState(id);
```

Comment:
```suggestion
            const state = await arweaveDataProvider.getContractState<PDNTContractJSON>(id);
```

and remove the case below

Reply:
removed / switch in https://github.com/ar-io/pdns-react/pull/212/commits/dbd3df90db10dcc9610fa630ef02e2c98b92a59e